src/sprites/setup.py

```diff
@@ -0,0 +1,257 @@
+
+    def get_hitbox(
+            self, state: EntityState, direction: Direction
+    ) -> pygame.Rect:
+        state_exception = self._state_exceptions.get(state)
+        direction_exception = self._direction_exceptions.get(direction)
+        exception = self._exceptions.get((state, direction))
+        if not exception:
+            if not direction_exception:
+                if not state_exception:
+                    return self.default
+                return state_exception
+            return direction_exception
+        return exception
+
+    def scale_hitboxes(self, factor: float) -> None:
+        for item in [self._state_exceptions.values(),
+                     self._direction_exceptions.values(),
+                     self._exceptions.values(),
+                     [self.default]]:
+            for hitbox in item:
+                hitbox.update(
+                    hitbox.left * factor,
+                    hitbox.top * factor,
+                    hitbox.width * factor,
+                    hitbox.height * factor
+                )
+
+
+class EntityAssets:
+    CHICKEN: EntityAsset
+    COW: EntityAsset
+    RABBIT: EntityAsset
+
+    @classmethod
+    def setup(cls):
+        chicken_hitbox = _Hitbox(pygame.Rect(1, 11, 11, 3))
+        chicken_hitbox.set_direction_exception(
+            Direction.LEFT, pygame.Rect(4, 11, 11, 3)
+        )
+
+        chicken_asset = entity_importer(
+            path=resource_path("images/entities/chicken"),
+            size=16,
+            directions=[Direction.RIGHT],
+            hitbox=chicken_hitbox
+        )
+
+        cls.CHICKEN = EntityAsset(
+            chicken_asset
+        )
+
+        cow_hitbox = _Hitbox(pygame.Rect(6, 25, 16, 4))
+        cow_hitbox.set_direction_exception(
+            Direction.LEFT, pygame.Rect(10, 25, 16, 4)
+        )
+
+        cow_asset = entity_importer(
+            path=resource_path("images/entities/cow"),
+            size=32,
+            directions=[Direction.RIGHT],
+            hitbox=cow_hitbox
+        )
+
+        cls.COW = EntityAsset(
+            cow_asset
+        )
+
+        rabbit_hitbox = _Hitbox(pygame.Rect(18, 26, 12, 6))
+        rabbit_hitbox.set_direction_exception(
+            Direction.LEFT, pygame.Rect(20, 26, 8, 6)
+        )
+        rabbit_hitbox.set_direction_exception(
+            Direction.RIGHT, pygame.Rect(20, 26, 8, 6)
+        )
+        rabbit_hitbox.set_exception(
+            EntityState.AXE, Direction.LEFT, pygame.Rect(24, 26, 8, 6)
+        )
+        rabbit_hitbox.set_exception(
+            EntityState.AXE, Direction.RIGHT, pygame.Rect(16, 26, 8, 6)
+        )
+
+        rabbit_asset = entity_importer(
+            path=resource_path("images/characters/rabbit"),
+            size=CHAR_TILE_SIZE,
+            directions=[Direction.DOWN, Direction.UP, Direction.LEFT],
+            hitbox=rabbit_hitbox
+        )
+
+        cls.RABBIT = EntityAsset(
+            rabbit_asset
+        )
```

Comment:
Again, that's a nitpick, but I feel `EntityAssets` could be just a namespace rather than a class, since no instance of it will exist.

In turn, `EntityAssets.setup` could become a global function `setup_entity_assets` that acts on said namespace.

Have a look at the docs for more information:

https://docs.python.org/3/library/types.html#types.SimpleNamespace

```suggestion
from types import SimpleNamespace

ENTITY_ASSETS = SimpleNamespace()

    def setup_entity_assets():
        chicken_hitbox = _Hitbox(pygame.Rect(1, 11, 11, 3))
        chicken_hitbox.set_direction_exception(
            Direction.LEFT, pygame.Rect(4, 11, 11, 3)
        )

        chicken_asset = entity_importer(
            path=resource_path("images/entities/chicken"),
            size=16,
            directions=[Direction.RIGHT],
            hitbox=chicken_hitbox
        )

        ENTITY_ASSETS.CHICKEN = EntityAsset(
            chicken_asset
        )

        cow_hitbox = _Hitbox(pygame.Rect(6, 25, 16, 4))
        cow_hitbox.set_direction_exception(
            Direction.LEFT, pygame.Rect(10, 25, 16, 4)
        )

        cow_asset = entity_importer(
            path=resource_path("images/entities/cow"),
            size=32,
            directions=[Direction.RIGHT],
            hitbox=cow_hitbox
        )

        ENTITY_ASSETS.COW = EntityAsset(
            cow_asset
        )

        rabbit_hitbox = _Hitbox(pygame.Rect(18, 26, 12, 6))
        rabbit_hitbox.set_direction_exception(
            Direction.LEFT, pygame.Rect(20, 26, 8, 6)
        )
        rabbit_hitbox.set_direction_exception(
            Direction.RIGHT, pygame.Rect(20, 26, 8, 6)
        )
        rabbit_hitbox.set_exception(
            EntityState.AXE, Direction.LEFT, pygame.Rect(24, 26, 8, 6)
        )
        rabbit_hitbox.set_exception(
            EntityState.AXE, Direction.RIGHT, pygame.Rect(16, 26, 8, 6)
        )

        rabbit_asset = entity_importer(
            path=resource_path("images/characters/rabbit"),
            size=CHAR_TILE_SIZE,
            directions=[Direction.DOWN, Direction.UP, Direction.LEFT],
            hitbox=rabbit_hitbox
        )

        ENTITY_ASSETS.RABBIT = EntityAsset(
            rabbit_asset
        )
```